Mathlib/Analysis/Complex/AbelLimit.lean

```diff
@@ -0,0 +1,212 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Analysis.Complex.Basic
+import Mathlib.Analysis.SpecificLimits.Normed
+
+/-!
+# Abel's limit theorem
+
+If a real or complex power series for a function has radius of convergence 1 and the series is only
+known to converge conditionally at 1, Abel's limit theorem gives the value at 1 as the limit of the
+function at 1 from the left. "Left" for complex numbers means "within a Stolz set", a subset of the
+open unit disc with 1 on its boundary.
+
+## Main theorems
+
+* `Complex.tendsto_tsum_powerSeries_nhdsWithin_stolzSet`:
+  Abel's limit theorem for complex power series.
+* `Real.tendsto_tsum_powerSeries_nhdsWithin_lt`: Abel's limit theorem for real power series.
+
+## References
+
+* https://planetmath.org/proofofabelslimittheorem
+* https://en.wikipedia.org/wiki/Abel%27s_theorem
+-/
+
+
+open Filter Finset
+
+open scoped BigOperators Topology
+
+namespace Complex
+
+section StolzSet
+
+/-- The Stolz set for a given `M`, roughly teardrop-shaped with the tip at 1 but tending to the
+open unit disc as `M` tends to infinity. -/
+def stolzSet (M : ℝ) : Set ℂ := {z | ‖z‖ < 1 ∧ ‖1 - z‖ < M * (1 - ‖z‖)}
+
+variable {M : ℝ}
+
+theorem stolzSet_empty (hM : M ≤ 1) : stolzSet M = ∅ := by
+  ext z
+  rw [stolzSet, Set.mem_setOf, Set.mem_empty_iff_false, iff_false, not_and, not_lt, ← sub_pos]
+  intro zn
+  calc
+    _ ≤ 1 * (1 - ‖z‖) := mul_le_mul_of_nonneg_right hM zn.le
+    _ = ‖(1 : ℂ)‖ - ‖z‖ := by rw [one_mul, norm_one]
+    _ ≤ _ := norm_sub_norm_le _ _
+
+theorem nhdsWithin_lt_le_nhdsWithin_stolzSet (hM : 1 < M) :
+    (𝓝[<] 1).map ofReal' ≤ 𝓝[stolzSet M] 1 := by
+  rw [← tendsto_id']
+  refine' tendsto_map' <| tendsto_nhdsWithin_of_tendsto_nhds_of_eventually_within ofReal'
+    (tendsto_nhdsWithin_of_tendsto_nhds <| ofRealCLM.continuous.tendsto' 1 1 rfl) _
+  simp only [eventually_iff, norm_eq_abs, abs_ofReal, abs_lt, mem_nhdsWithin]
+  refine' ⟨Set.Ioo 0 2, isOpen_Ioo, by norm_num, fun x hx ↦ _⟩
+  simp only [Set.mem_inter_iff, Set.mem_Ioo, Set.mem_Iio] at hx
+  simp only [Set.mem_setOf_eq, stolzSet, ← ofReal_one, ← ofReal_sub, norm_eq_abs, abs_ofReal,
+    abs_of_pos hx.1.1, abs_of_pos <| sub_pos.mpr hx.2]
+  exact ⟨hx.2, lt_mul_left (sub_pos.mpr hx.2) hM⟩
+
+end StolzSet
+
+variable {f : ℕ → ℂ} {l : ℂ} (h : Tendsto (fun n ↦ ∑ i in range n, f i) atTop (𝓝 l))
```

Comment:
Please move `h` to the statements, otherwise they're impossible to understand in isolation.

Reply:
Done.